packages/api/src/command/medical/patient/hl7-fhir-webhook.ts

```diff
@@ -1,14 +1,73 @@
-import { out } from "@metriport/core/util";
+import { capture, out } from "@metriport/core/util";
+import { uuidv7 } from "@metriport/core/util/uuid-v7";
+import { MetriportError, errorToString } from "@metriport/shared";
+import { getSettingsOrFail } from "../../settings/getSettings";
+import { processRequest } from "../../webhook/webhook";
+import { createWebhookRequest } from "../../webhook/webhook-request";
+import { getPatientOrFail } from "./get-patient";
 
-// TODO 2888: Implement this function
 export async function processHl7FhirBundleWebhook({
   cxId,
   patientId,
+  presignedUrl,
+  triggerEvent,
 }: {
   cxId: string;
   patientId: string;
   presignedUrl: string;
+  triggerEvent: string;
 }): Promise<void> {
   const { log } = out(`processHl7FhirBundleWebhook, cx: ${cxId}, pt: ${patientId}`);
-  log(`Will process presignedUrl... [STUB]`);
+  const requestId = uuidv7();
+  log(`req - ${requestId}, event - ${triggerEvent}`);
+  try {
+    const [settings, currentPatient] = await Promise.all([
+      getSettingsOrFail({ id: cxId }),
+      getPatientOrFail({ id: patientId, cxId }),
+    ]);
+    const webhookType = mapTriggerEventToWebhookType(triggerEvent);
+
+    const payload = {
+      payload: {
+        patientId,
+        ...(currentPatient.externalId ? { externalId: currentPatient.externalId } : {}),
+        ...(currentPatient.additionalIds ? { additionalIds: currentPatient.additionalIds } : {}),
+        url: presignedUrl,
+      },
+    };
+
+    const webhookRequest = await createWebhookRequest({
+      cxId,
+      type: `medical.hl7.${webhookType}`,
+      payload,
+      requestId,
+    });
+
+    const additionalWHRequestMeta: Record<string, string> = { requestId };
+
+    await processRequest(webhookRequest, settings, additionalWHRequestMeta);
+  } catch (err) {
+    const msg = `Failed to send hl7 notification webhook`;
+    log(`${msg}, error - ${errorToString(err)}`);
+    capture.error(msg, {
+      extra: { patientId, context: `webhook.processHl7FhirBundleWebhook`, err },
+    });
+    throw err;
+  }
+}
+
+function mapTriggerEventToWebhookType(triggerEvent: string) {
+  switch (triggerEvent) {
+    case "A01": {
+      return "admit";
+    }
+    case "A03": {
+      return "discharge";
+    }
+    default: {
+      throw new MetriportError("Unsupported HL7 triggerEvent for the webhook", undefined, {
+        triggerEvent,
+      });
+    }
+  }
 }
```

Comment:
The above looks good, I often try to avoid switch statements because sometimes people want to start shoving logic in between those brackets.

This is not even a suggestion lol, just more of a comment.

```
function mapTriggerEventToWebhookType(triggerEvent: string) {
  const whType = {
    "A01": "admit" as const,
    "A03": "discharge" as const
  }[triggerEvent]
  
  if (!whType) {  
      throw new MetriportError("Unsupported HL7 triggerEvent for the webhook", undefined, {
        triggerEvent,
      });

  return whType
}
```